R/glmCommonFunctions.R

```diff
@@ -55,34 +55,70 @@
   # make formulas for the full model and the null model
   ff <- .createGLMFormula(options, nullModel = FALSE)
   nf <- .createGLMFormula(options, nullModel = TRUE)
-  # specify family and link
-  if (options$family == "bernoulli") {
-    family <- "binomial"
-  }
-  else if (options$family == "gamma") {
-    family <- "Gamma"
-  }
-  else if (options$family == "inverseGaussian") {
-    family <- "inverse.gaussian"
-  }
-  else {
-    family <- options$family
-  }
 
-  familyLink <- eval(call(family, link = options$link))
-  # compute full and null models
-  if (options$weights == "") {
-    fullModel <- stats::glm(ff, family = familyLink, data = dataset, weights = NULL)
-    nullModel <- stats::glm(nf, family = familyLink, data = dataset, weights = NULL)
-  } else {
+  if (options$family != "other") {
+    # specify family and link
+    if (options$family == "bernoulli") {
+      family <- "binomial"
+    }
+    else if (options$family == "gamma") {
+      family <- "Gamma"
+    }
+    else if (options$family == "inverseGaussian") {
+      family <- "inverse.gaussian"
+    }
+    else {
+      family <- options$family
+    }
+
+    familyLink <- eval(call(family, link = options$link))
+    # compute full and null models
     fullModel <- stats::glm(ff,
                             family = familyLink,
                             data = dataset,
-                            weights = get(options$weights))
+                            weights = eval(.getWeightVariable(options$weights)))
     nullModel <- stats::glm(nf,
                             family = familyLink,
                             data = dataset,
-                            weights = get(options$weights))
+                            weights = eval(.getWeightVariable(options$weights)))
+  }
+
+  if (options$family == "other") {
```

Comment:
this should probably be `else if`

Reply:
I think `if` is fine in this case.

Reply:
Is there a preference for using `else if`?

Reply:
for `options$family`? Could also be an else?
now you're doing
```r
if (options$family != "other") {
  ... # code
} 

if (options$family == "other") {
  ... # code
} 
```
but if the two code paths are mutually exclusive then this is more clear:
```r
if (options$family != "other") {
  ... # code
} else { # family == "other"
  ... # code
} 
```
in the original code it's not clear that the two code paths are mutually exclusive (since the first one may change the value of `options$family`).

Reply:
true, I'll fix that. thx